### DISCUSSION TITLE
Node Attributes Prediction

### DISCUSSION THREAD

**Discussion by imayachita** [danielegrattarola/spektral#177] [Q&A]
Hi @danielegrattarola,
I am wondering if I can use Spektral to perform node attributes prediction instead of node classification?

And also, is it possible that X (node attributes matrix) is 3-D instead of 2-D? For example: we want to use word embedding as the node attributes instead of bag of words representation. If using bag of words, X is [num of nodes, num of words in the vocab]. If using word embedding, X will be similar to sequence models input, which is [num of nodes, max sequence length, num of word embedding dimension].

Thanks!

**Comment by danielegrattarola** [comment]
Hi, 

you mean node regression, where you learn to predict a continuous value and not a label? If so, yes. You just need to change the final activation function and the loss, just like you would do in any other neural network. 

Regarding a 3D attributes matrix, it should be possible, in principle, to create a GNN layer that deals with it, but Spektral does not support it for now. Usually, we have a 3D node attributes matrix when we are in [batch mode](https://graphneural.network/data/#batch-mode), and the code often checks for that.

Cheers
